Gmission prototype

jhgan00 opened the pull request:
- 패치코어 모듈 추가
-`patch_core.py`, `main.py` 중점적으로 봐주시면 될것같아용 ~

jhgan00 committed:
후처리 수정 & conf 반환

jhgan00 committed:
updated yolo pre/post-processing

- use aspectaware resizing (letterbox function in utils.py)

jhgan00 committed:
updated segmentation model

- from: unet -> to: deeplab
- updated roi preprocessing

jhgan00 committed:
added anomaly detection stage

- 넙치 영역 분류기 추가(전체 여부)
- PatchCore 모델 추가

jhgan00 committed:
updated comments & configs & env vars

- 코드 리뷰를 위해 코드 정리 & 주석 추가

dhlee-work commented on main.py:
```diff
@@ -94,29 +97,32 @@ async def get_bboxes_and_diseases(file: UploadFile) -> JSONResponse:
     classification_rois /= IMAGENET_STD
-    mask = CLASSIFICATION_SESSION.batch_run(classification_rois, config.classification.batch_size).flatten() > 0
+    logits = CLASSIFICATION_SESSION.batch_run(classification_rois, config.classification.batch_size).flatten()
+    mask = logits > 0  # sigmoid 생략: 0 기준으로 몸통 전체 여부 판단
```
차후 임계값을 조절할 필요가 있으므로 sigmoid로 두고 config에서 설정할 수 있도록 하면 좋겠음

jhgan00 committed:
코드 리뷰 반영

- 분류모델 출력을 시그모이드로 변경하고 설정파일에 임계값 추가
- anomaly_score, is_whole_body 정보 추가
- pad_and_resize -> letterbox 함수로 통일
- 테스트 스크립트 수정: 변수 사용

jhgan00 commented on main.py:
```diff
@@ -47,58 +58,86 @@ async def get_bboxes_and_diseases(file: UploadFile) -> JSONResponse:
+    classification_rois /= IMAGENET_STD
+    logits = CLASSIFICATION_SESSION.batch_run(classification_rois, config.classification.batch_size).flatten()
+    probs = sigmoid(logits)
+    mask = probs > config.classification.threshold  # sigmoid & threshold 적용하도록 수정
```
시그모이드 적용하였습니다 ~

jhgan00 commented on config/gmission.proto.yml:
```diff
@@ -1,22 +1,34 @@
+  input_size: 128
   device_id: 0
   batch_size: 8
+  threshold: 0.5
```
시그모이드 threshold 설정 추가
